From bc20e87822c3031d65c63467a1d86f4f5f24b31a Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Fri, 11 Sep 2015 13:52:49 +0800
Subject: [PATCH 52/53] arm: dts: zynq-zc702: add support TRD

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

This patch is derived out of the file rdf0286-zc702-zvik-base-trd-2014-2.zip
which downloaded from Xilinx website.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/zynq-zc702-base-trd.dts |  723 +++++++++++++++++++++++++++++
 1 file changed, 723 insertions(+)
 create mode 100644 arch/arm/boot/dts/zynq-zc702-base-trd.dts

diff --git a/arch/arm/boot/dts/zynq-zc702-base-trd.dts b/arch/arm/boot/dts/zynq-zc702-base-trd.dts
new file mode 100644
index 0000000..bd30918
--- /dev/null
+++ b/arch/arm/boot/dts/zynq-zc702-base-trd.dts
@@ -0,0 +1,723 @@
+/*
+ * Device Tree Generator version: 1.0
+ *
+ * (C) Copyright 2015 WindRiver, Inc.
+ *
+ */
+
+/dts-v1/;
+
+/ {
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	compatible = "xlnx,zynq-7000";
+	model = "Xilinx Zynq";
+
+	aliases {
+		ethernet0 = "/amba@0/ps7-ethernet@e000b000";
+		i2c0 = "/amba@0/ps7-i2c@e0004000";
+		i2c1 = "/amba@0/ps7-i2c@e0005000";
+		serial0 = "/amba@0/serial@e0001000";
+		spi0 = "/amba@0/ps7-qspi@e000d000";
+	};
+
+	chosen {
+		bootargs = "console=ttyPS0,115200 root=/dev/ram rw earlyprintk ramdisk_size=32768 consoleblank=0";
+		linux,stdout-path = "/amba@0/serial@e0001000";
+	};
+
+	cpus {
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+
+		cpu@0 {
+			bus-handle = <0x1>;
+			clock-latency = <0x3e8>;
+			clocks = <0x2 0x3>;
+			compatible = "arm,cortex-a9";
+			device_type = "cpu";
+			interrupt-handle = <0x3>;
+			operating-points = <0xa2c2b 0xf4240 0x51616 0xf4240 0x3640f 0xf4240>;
+			reg = <0x0>;
+		};
+
+		cpu@1 {
+			bus-handle = <0x1>;
+			clocks = <0x2 0x3>;
+			compatible = "arm,cortex-a9";
+			device_type = "cpu";
+			interrupt-handle = <0x3>;
+			reg = <0x1>;
+		};
+	};
+
+	pmu {
+		compatible = "arm,cortex-a9-pmu";
+		interrupt-parent = <0x3>;
+		interrupts = <0x0 0x5 0x4 0x0 0x6 0x4>;
+		reg = <0xf8891000 0x1000 0xf8893000 0x1000>;
+		reg-names = "cpu0", "cpu1";
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x40000000>;
+	};
+
+	amba@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "xlnx,ps7-axi-interconnect-1.00.a", "simple-bus";
+		ranges;
+		linux,phandle = <0x1>;
+		phandle = <0x1>;
+
+		ps7-afi@f8008000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = <0xf8008000 0x1000>;
+		};
+
+		ps7-afi@f8009000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = <0xf8009000 0x1000>;
+		};
+
+		ps7-afi@f800a000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = <0xf800a000 0x1000>;
+		};
+
+		ps7-afi@f800b000 {
+			compatible = "xlnx,ps7-afi-1.00.a";
+			reg = <0xf800b000 0x1000>;
+		};
+
+		ps7-can@e0008000 {
+			clock-names = "ref_clk", "aper_clk";
+			clocks = <0x2 0x13 0x2 0x24>;
+			compatible = "xlnx,zynq-can-1.0";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x1c 0x4>;
+			reg = <0xe0008000 0x1000>;
+		};
+
+		ps7-ddrc@f8006000 {
+			compatible = "xlnx,zynq-ddrc-1.0";
+			reg = <0xf8006000 0x1000>;
+			xlnx,has-ecc = <0x0>;
+		};
+
+		ps7-dev-cfg@f8007000 {
+			clock-names = "ref_clk", "fclk0", "fclk1", "fclk2", "fclk3";
+			clocks = <0x2 0xc 0x2 0xf 0x2 0x10 0x2 0x11 0x2 0x12>;
+			compatible = "xlnx,zynq-devcfg-1.0";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x8 0x4>;
+			reg = <0xf8007000 0x100>;
+		};
+
+		ps7-dma@f8003000 {
+			#dma-cells = <0x1>;
+			#dma-channels = <0x8>;
+			#dma-requests = <0x4>;
+			clock-names = "apb_pclk";
+			clocks = <0x2 0x1b>;
+			compatible = "arm,primecell", "arm,pl330";
+			interrupt-names = "abort", "dma0", "dma1", "dma2", "dma3", "dma4", "dma5", "dma6", "dma7";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0xd 0x4 0x0 0xe 0x4 0x0 0xf 0x4 0x0 0x10 0x4 0x0 0x11 0x4 0x0 0x28 0x4 0x0 0x29 0x4 0x0 0x2a 0x4 0x0 0x2b 0x4>;
+			reg = <0xf8003000 0x1000>;
+		};
+
+		ps7-ethernet@e000b000 {
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			clock-names = "ref_clk", "aper_clk";
+			clocks = <0x2 0xd 0x2 0x1e>;
+			compatible = "xlnx,ps7-ethernet-1.00.a";
+			enet-reset = <0x4 0xb 0x0>;
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x16 0x4>;
+			local-mac-address = [00 0a 35 00 00 00];
+			phy-handle = <0x5>;
+			phy-mode = "rgmii-id";
+			reg = <0xe000b000 0x1000>;
+			xlnx,eth-mode = <0x1>;
+			xlnx,has-mdio = <0x1>;
+			xlnx,ptp-enet-clock = <0x69f6bcb>;
+
+			mdio {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+
+				phy@7 {
+					compatible = "marvell,88e1116r";
+					device_type = "ethernet-phy";
+					reg = <0x7>;
+					linux,phandle = <0x5>;
+					phandle = <0x5>;
+				};
+			};
+		};
+
+		ps7-globaltimer@f8f00200 {
+			clocks = <0x2 0x4>;
+			compatible = "arm,cortex-a9-global-timer";
+			interrupt-parent = <0x3>;
+			interrupts = <0x1 0xb 0x301>;
+			reg = <0xf8f00200 0x100>;
+		};
+
+		ps7-gpio@e000a000 {
+			#gpio-cells = <0x2>;
+			clocks = <0x2 0x2a>;
+			compatible = "xlnx,zynq-gpio-1.0";
+			emio-gpio-width = <0x40>;
+			gpio-controller;
+			gpio-mask-high = <0x0>;
+			gpio-mask-low = <0x5600>;
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x14 0x4>;
+			reg = <0xe000a000 0x1000>;
+			linux,phandle = <0x4>;
+			phandle = <0x4>;
+		};
+
+		ps7-i2c@e0004000 {
+			clock-frequency = <0x61a80>;
+			clocks = <0x2 0x26>;
+			compatible = "cdns,i2c-r1p10";
+			i2c-reset = <0x4 0xd 0x0>;
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x19 0x4>;
+			reg = <0xe0004000 0x1000>;
+			xlnx,has-interrupt = <0x0>;
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+
+			i2cswitch@74 {
+				compatible = "nxp,pca9548";
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				reg = <0x74>;
+
+				i2c@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x0>;
+					reg = <0x0>;
+
+					clock-generator@5d {
+						#clock-cells = <0x0>;
+						compatible = "silabs,si570";
+						temperature-stability = <0x32>;
+						reg = <0x5d>;
+						factory-fout = <0x9502f90>;
+						clock-frequency = <0x8d9ee20>;
+						linux,phandle = <0xf>;
+						phandle = <0xf>;
+					};
+				};
+
+				i2c@1 {
+					#size-cells = <0x0>;
+					#address-cells = <0x1>;
+					reg = <0x1>;
+
+					adv7511@39 {
+						compatible = "adi,adv7511";
+						reg = <0x39>;
+						adi,input-style = <0x3>;
+						adi,input-id = <0x1>;
+						adi,input-color-depth = <0x3>;
+						adi,sync-pulse = <0x3>;
+						adi,bit-justification = <0x1>;
+						adi,up-conversion = <0x0>;
+						adi,timing-generation-sequence = <0x0>;
+						adi,vsync-polarity = <0x2>;
+						adi,hsync-polarity = <0x2>;
+						adi,tdms-clock-inversion;
+						adi,clock-delay = <0x0>;
+						linux,phandle = <0x11>;
+						phandle = <0x11>;
+					};
+				};
+
+				i2c@2 {
+					#address-cells = <0x1>;
+					#size-cells = <0x0>;
+					reg = <0x2>;
+
+					eeprom@54 {
+						compatible = "at,24c08";
+						reg = <0x54>;
+					};
+				};
+
+				i2c@3 {
+					#address-cells = <0x1>;
+					#size-cells = <0x0>;
+					reg = <0x3>;
+
+					gpio@21 {
+						compatible = "ti,tca6416";
+						reg = <0x21>;
+						gpio-controller;
+						#gpio-cells = <0x2>;
+					};
+				};
+
+				i2c@4 {
+					#address-cells = <0x1>;
+					#size-cells = <0x0>;
+					reg = <0x4>;
+
+					rtc@51 {
+						compatible = "nxp,pcf8563";
+						reg = <0x51>;
+					};
+				};
+
+				i2c@7 {
+					#address-cells = <0x1>;
+					#size-cells = <0x0>;
+					reg = <0x7>;
+
+					hwmon@52 {
+						compatible = "ti,ucd9248";
+						reg = <0x34>;
+					};
+
+					hwmon@53 {
+						compatible = "ti,ucd9248";
+						reg = <0x35>;
+					};
+
+					hwmon@54 {
+						compatible = "ti,ucd9248";
+						reg = <0x36>;
+					};
+				};
+			};
+		};
+
+		ps7-i2c@e0005000 {
+			clock-frequency = <0x61a80>;
+			clocks = <0x2 0x27>;
+			compatible = "cdns,i2c-r1p10";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x30 0x4>;
+			reg = <0xe0005000 0x1000>;
+			xlnx,has-interrupt = <0x0>;
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+		};
+
+		ps7-iop-bus-config@e0200000 {
+			compatible = "xlnx,ps7-iop-bus-config-1.00.a";
+			reg = <0xe0200000 0x1000>;
+		};
+
+		ps7-ocmc@f800c000 {
+			compatible = "xlnx,zynq-ocmc-1.0";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x3 0x4>;
+			reg = <0xf800c000 0x1000>;
+		};
+
+		ps7-pl310@f8f02000 {
+			arm,data-latency = <0x3 0x2 0x2>;
+			arm,tag-latency = <0x2 0x2 0x2>;
+			cache-level = <0x2>;
+			cache-unified;
+			compatible = "arm,pl310-cache";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x2 0x4>;
+			reg = <0xf8f02000 0x1000>;
+		};
+
+		ps7-qspi@e000d000 {
+			clock-names = "ref_clk", "pclk";
+			clocks = <0x2 0xa 0x2 0x2b>;
+			compatible = "xlnx,zynq-qspi-1.0";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x13 0x4>;
+			is-dual = <0x0>;
+			num-cs = <0x1>;
+			reg = <0xe000d000 0x1000>;
+			xlnx,fb-clk = <0x1>;
+			xlnx,qspi-mode = <0x0>;
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+
+			flash@0 {
+				compatible = "n25q128";
+				reg = <0x0>;
+				spi-tx-bus-width = <0x1>;
+				spi-rx-bus-width = <0x4>;
+				spi-max-frequency = <0x2faf080>;
+				#address-cells = <0x1>;
+				#size-cells = <0x1>;
+
+				partition@qspi-fsbl-uboot {
+					label = "qspi-fsbl-uboot";
+					reg = <0x0 0x100000>;
+				};
+
+				partition@qspi-linux {
+					label = "qspi-linux";
+					reg = <0x100000 0x500000>;
+				};
+
+				partition@qspi-device-tree {
+					label = "qspi-device-tree";
+					reg = <0x600000 0x20000>;
+				};
+
+				partition@qspi-rootfs {
+					label = "qspi-rootfs";
+					reg = <0x620000 0x5e0000>;
+				};
+
+				partition@qspi-bitstream {
+					label = "qspi-bitstream";
+					reg = <0xc00000 0x400000>;
+				};
+			};
+		};
+
+		ps7-qspi-linear@fc000000 {
+			clock-names = "ref_clk", "aper_clk";
+			clocks = <0x2 0xa 0x2 0x2b>;
+			compatible = "xlnx,ps7-qspi-linear-1.00.a";
+			reg = <0xfc000000 0x1000000>;
+		};
+
+		ps7-scugic@f8f01000 {
+			#address-cells = <0x2>;
+			#interrupt-cells = <0x3>;
+			#size-cells = <0x1>;
+			compatible = "arm,cortex-a9-gic", "arm,gic";
+			interrupt-controller;
+			num_cpus = <0x2>;
+			num_interrupts = <0x60>;
+			reg = <0xf8f01000 0x1000 0xf8f00100 0x100>;
+			linux,phandle = <0x3>;
+			phandle = <0x3>;
+		};
+
+		ps7-scutimer@f8f00600 {
+			clocks = <0x2 0x4>;
+			compatible = "arm,cortex-a9-twd-timer";
+			interrupt-parent = <0x3>;
+			interrupts = <0x1 0xd 0x301>;
+			reg = <0xf8f00600 0x20>;
+		};
+
+		ps7-scuwdt@f8f00620 {
+			clocks = <0x2 0x4>;
+			compatible = "xlnx,ps7-scuwdt-1.00.a";
+			device_type = "watchdog";
+			interrupt-parent = <0x3>;
+			interrupts = <0x1 0xe 0x301>;
+			reg = <0xf8f00620 0xe0>;
+		};
+
+		ps7_sd_0: ps7-sdio@e0100000 {
+			clock-frequency = <50000000>;
+			clock-names = "ref_clk", "aper_clk";
+			clocks = <0x02 21>, <0x02 32>;
+			compatible = "xlnx,ps7-sdio-1.00.a", "generic-sdhci", "arasan,sdhci";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x18 0x4>;
+			reg = <0xe0100000 0x1000>;
+			xlnx,has-cd = <0x1>;
+			xlnx,has-power = <0x0>;
+			xlnx,has-wp = <0x1>;
+		};
+
+		ps7-slcr@f8000000 {
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			compatible = "xlnx,zynq-slcr", "syscon";
+			ranges;
+			reg = <0xf8000000 0x1000>;
+			clkc@100 {
+				#clock-cells = <0x1>;
+				clock-output-names = "armpll", "ddrpll", "iopll", "cpu_6or4x", "cpu_3or2x", "cpu_2x", "cpu_1x", "ddr2x", "ddr3x", "dci", "lqspi", "smc", "pcap", "gem0", "gem1", "fclk0", "fclk1", "fclk2", "fclk3", "can0", "can1", "sdio0", "sdio1", "uart0", "uart1", "spi0", "spi1", "dma", "usb0_aper", "usb1_aper", "gem0_aper", "gem1_aper", "sdio0_aper", "sdio1_aper", "spi0_aper", "spi1_aper", "can0_aper", "can1_aper", "i2c0_aper", "i2c1_aper", "uart0_aper", "uart1_aper", "gpio_aper", "lqspi_aper", "smc_aper", "swdt", "dbg_trc", "dbg_apb";
+				compatible = "xlnx,ps7-clkc";
+				fclk-enable = <0xf>;
+				ps-clk-frequency = <0x1fca055>;
+				reg = <0x100 0x100>;
+				linux,phandle = <0x2>;
+				phandle = <0x2>;
+			};
+		};
+
+		ps7-ttc@f8001000 {
+			clocks = <0x2 0x6>;
+			compatible = "cdns,ttc";
+			interrupt-names = "ttc0", "ttc1", "ttc2";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0xa 0x4 0x0 0xb 0x4 0x0 0xc 0x4>;
+			reg = <0xf8001000 0x1000>;
+		};
+
+		serial@e0001000 {
+			clock-names = "ref_clk", "aper_clk";
+			clocks = <0x2 0x18 0x2 0x29>;
+			compatible = "xlnx,xuartps";
+			current-speed = <0x1c200>;
+			device_type = "serial";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x32 0x4>;
+			port-number = <0x0>;
+			reg = <0xe0001000 0x1000>;
+			xlnx,has-modem = <0x0>;
+		};
+
+		ps7-usb@e0002000 {
+			clocks = <0x2 0x1c>;
+			compatible = "xlnx,ps7-usb-1.00.a", "xlnx,zynq-usb-1.00.a";
+			dr_mode = "host";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x15 0x4>;
+			phy_type = "ulpi";
+			reg = <0xe0002000 0x1000>;
+			usb-reset = <0x4 0x7 0x0>;
+		};
+
+		ps7-wdt@f8005000 {
+			clocks = <0x2 0x2d>;
+			compatible = "xlnx,zynq-wdt-r1p2";
+			device_type = "watchdog";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x9 0x1>;
+			reg = <0xf8005000 0x1000>;
+			reset = <0x0>;
+			timeout-sec = <0xa>;
+		};
+
+		ps7-xadc@f8007100 {
+			clocks = <0x2 0xc>;
+			compatible = "xlnx,zynq-xadc-1.00.a";
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x7 0x4>;
+			reg = <0xf8007100 0x20>;
+		};
+
+		axi-perf-mon@400f0000 {
+			compatible = "generic-uio";
+			reg = <0x400f0000 0x10000>;
+		};
+
+		vtc@40070000 {
+			compatible = "xlnx,axi-vtc-6.1";
+			reg = <0x40070000 0x10000>;
+			clocks = <0x2 0xf>;
+			xlnx,generator;
+			linux,phandle = <0x6>;
+			phandle = <0x6>;
+		};
+
+		axi_tpg@40080000 {
+			compatible = "xlnx,axi-tpg-5.0";
+			reg = <0x40080000 0x10000>;
+			xlnx,vtc = <0x6>;
+			timing-gpios = <0x4 0x37 0x1>;
+
+			ports {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+
+				port@0 {
+					reg = <0x0>;
+					xlnx,axi-video-format = "yuv422";
+					xlnx,axi-video-width = <0x8>;
+				};
+
+				port@1 {
+					reg = <0x1>;
+					xlnx,axi-video-format = "yuv422";
+					xlnx,axi-video-width = <0x8>;
+
+					endpoint {
+						remote-endpoint = <0x7>;
+						linux,phandle = <0x9>;
+						phandle = <0x9>;
+					};
+				};
+			};
+		};
+
+		axivdma@40090000 {
+			compatible = "xlnx,axi-vdma";
+			reg = <0x40090000 0x10000>;
+			xlnx,flush-fsync = <0x1>;
+			xlnx,num-fstores = <0x1>;
+			#dma-cells = <0x1>;
+			linux,phandle = <0x8>;
+			phandle = <0x8>;
+
+			dma-s2mmchannel@40090000 {
+				compatible = "xlnx,axi-vdma-s2mm-channel";
+				interrupt-parent = <0x3>;
+				interrupts = <0x0 0x39 0x4>;
+				xlnx,datawidth = <0x40>;
+				xlnx,device-id = <0x0>;
+			};
+		};
+
+		axi_video_cap {
+			compatible = "xlnx,axi-video";
+			dmas = <0x8 0x0>;
+			dma-names = "port0";
+
+			ports {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+
+				port@0 {
+					reg = <0x0>;
+					direction = "input";
+
+					endpoint {
+						remote-endpoint = <0x9>;
+						linux,phandle = <0x7>;
+						phandle = <0x7>;
+					};
+				};
+			};
+		};
+
+		axi_hls@400d0000 {
+			compatible = "xlnx,axi-hls-sobel", "xlnx,axi-hls";
+			reg = <0x400d0000 0x24 0x400d0024 0xa8>;
+
+			ports {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+
+				port@0 {
+					reg = <0x0>;
+					xlnx,axi-video-format = "yuv422";
+					xlnx,axi-video-width = <0x8>;
+
+					endpoint {
+						remote-endpoint = <0xa>;
+						linux,phandle = <0xe>;
+						phandle = <0xe>;
+					};
+				};
+
+				port@1 {
+					reg = <0x1>;
+					xlnx,axi-video-format = "yuv422";
+					xlnx,axi-video-width = <0x8>;
+
+					endpoint {
+						remote-endpoint = <0xb>;
+						linux,phandle = <0xd>;
+						phandle = <0xd>;
+					};
+				};
+			};
+		};
+
+		axivdma@400b0000 {
+			compatible = "xlnx,axi-vdma";
+			reg = <0x400b0000 0x10000>;
+			xlnx,flush-fsync = <0x1>;
+			xlnx,num-fstores = <0x1>;
+			#dma-cells = <0x1>;
+			linux,phandle = <0xc>;
+			phandle = <0xc>;
+
+			dma-mm2schannel@400b0000 {
+				compatible = "xlnx,axi-vdma-mm2s-channel";
+				interrupt-parent = <0x3>;
+				interrupts = <0x0 0x37 0x4>;
+				xlnx,datawidth = <0x40>;
+				xlnx,device-id = <0x0>;
+			};
+
+			dma-s2mmchannel@0x400b0000 {
+				compatible = "xlnx,axi-vdma-s2mm-channel";
+				interrupt-parent = <0x3>;
+				interrupts = <0x0 0x38 0x4>;
+				xlnx,datawidth = <0x40>;
+				xlnx,device-id = <0x1>;
+			};
+		};
+
+		axi_video_m2m {
+			compatible = "xlnx,axi-video";
+			dmas = <0xc 0x1 0xc 0x0>;
+			dma-names = "port0", "port1";
+
+			ports {
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+
+				port@0 {
+					reg = <0x0>;
+					direction = "input";
+
+					endpoint {
+						remote-endpoint = <0xd>;
+						linux,phandle = <0xb>;
+						phandle = <0xb>;
+					};
+				};
+
+				port@1 {
+					reg = <0x1>;
+					direction = "output";
+
+					endpoint {
+						remote-endpoint = <0xe>;
+						linux,phandle = <0xa>;
+						phandle = <0xa>;
+					};
+				};
+			};
+		};
+
+		logicvc@40030000 {
+			compatible = "xylon,logicvc-4.00.a";
+			reg = <0x40030000 0x10000>;
+			interrupt-parent = <0x3>;
+			interrupts = <0x0 0x3a 0x4>;
+			background-layer-bits-per-pixel = <0x20>;
+			background-layer-type = "rgb";
+			color-space = <0x1>;
+			interface = <0x0>;
+			is-readable-regs;
+			is-size-position;
+			pixel-stride = <0x800>;
+			pixel-clock-active-high = <0x1>;
+			pixel-data-invert = <0x0>;
+			linux,phandle = <0x10>;
+			phandle = <0x10>;
+
+			layer_0 {
+				bits-per-pixel = <0x20>;
+				type = "rgb";
+				transparency = "layer";
+			};
+
+			layer_1 {
+				bits-per-pixel = <0x10>;
+				type = "yuv";
+				transparency = "layer";
+			};
+		};
+
+		xylon_drm {
+			compatible = "xylon,drm-1.00.a";
+			clocks = <0xf>;
+			device = <0x10>;
+			encoder = <0x11>;
+			private-plane = <0x0>;
+		};
+	};
+};
-- 
1.7.10.4

